packages/api/src/task/scheduler.ts

```diff
@@ -347,54 +347,50 @@ export class TaskScheduler {
   async createAndScheduleTask(
     type: Task["type"],
     params: Task["params"],
+    user: User,
     inputAsset?: Asset,
     outputAsset?: Asset,
     userId?: string,
     requesterId?: string,
   ) {
+    const projectId = inputAsset?.projectId || user?.defaultProjectId;
```

Comment:
Shouldn't this come from the request API key instead? i.e. `req.project?.id || user.defaultProjectId` (user is never null, should not `?`)

Reply:
Makes sense, but in the `clip` case the owner is different than the requester - I could pass req.project.id everywhere but there, but I instead removed userId and I am passing the user object, passing the owner during the clip gen, how does it look?

Reply:
Seeing a bunch of test failure after addressing the comments tho, looking into it